DataDog/integrations-core
# [CONTINT-477] Add rate_limit_queries_remaining_min  metric to default metrics in datadog_cluster_agent integration

### What does this PR do?

This PR adds a new metric (`rate_limit_queries_remaining`) to the cluster agent integration, reported as `datadog.cluster_agent.datadog.rate_limit_queries.remaining_min`. 

### Motivation

This metric can will allow users to create monitors and get alerted when the rate limit of queries sent to the backend by the external metrics server is clause to exhaustion.

- This new metric is exposed by the external metrics server, and is updated after every single query sent to the backend, as opposed to `datadog.cluster_agent.datadog.rate_limit_queries.remaining` which is updated only after the last query of a batch. 
- This new metric also keeps the minimum value of the rate limit remaining requests for at least `2*refresh_period` (refresh_period is set to 30 seconds by default).
- This new metric can help create monitors that are less flaky and more sensitive to rate limit errors, which can help improve taking action before rate limit error occurs.

### Additional Notes
- This metric is exposed by the cluster agent telemetry. It was added in [this PR](https://github.com/DataDog/datadog-agent/pull/20497).

### Testing
- The fixtures file and unit tests were updated for the `datadog_cluster_agent` integration. Running `ddev test datadog_cluster_agent` can be used to run the unit tests of this integration.
- QA
  - Deploy the agent/cluster agent on a kubernetes cluster with external metrics being enabled on the cluster agent and used in the cluster by creating some HPA/WPA and DatadogMetric objects (take [this](https://docs.datadoghq.com/containers/guide/cluster_agent_autoscaling_metrics/?tab=helm) as an example)  
  - Run the `datadog_cluster_agent` check and ensure that `datadog.cluster_agent.datadog.rate_limit_queries.remaining` appears in the output as in the following sample output:
```datadog.cluster_agent.datadog.rate_limit_queries.remaining      gauge  1699018474  1594                docker_image:cluster-agent:recent-1, endpoint:/api/v1/query, image_id:docker://sha256:924bfb95745e71f4f502e4f2f0449a74fbff935f829eecfdf84ae50541b6403f, image_name:cluster-agent, image_tag:recent-1, is_leader:true, join_leader:true, kube_app_component:cluster-agent, kube_app_instance:datadog-agent, kube_app_managed_by:Helm, kube_app_name:datadog-agent, kube_container_name:cluster-agent, kube_deployment:datadog-agent-cluster-agent, kube_namespace:default, kube_ownerref_kind:replicaset, kube_qos:BestEffort, kube_replica_set:datadog-agent-cluster-agent-7cd8d7f689, kube_service:datadog-agent-cluster-agent, kube_service:datadog-agent-cluster-agent-admission-controller, kube_service:datadog-agent-cluster-agent-metrics-api, pod_phase:running, short_image:cluster-agent                                                                         
  datadog.cluster_agent.datadog.rate_limit_queries.remaining_min  gauge  1699018474  1585                docker_image:cluster-agent:recent-1, endpoint:/api/v1/query, image_id:docker://sha256:924bfb95745e71f4f502e4f2f0449a74fbff935f829eecfdf84ae50541b6403f, image_name:cluster-agent, image_tag:recent-1, is_leader:true, join_leader:true, kube_app_component:cluster-agent, kube_app_instance:datadog-agent, kube_app_managed_by:Helm, kube_app_name:datadog-agent, kube_container_name:cluster-agent, kube_deployment:datadog-agent-cluster-agent, kube_namespace:default, kube_ownerref_kind:replicaset, kube_qos:BestEffort, kube_replica_set:datadog-agent-cluster-agent-7cd8d7f689, kube_service:datadog-agent-cluster-agent, kube_service:datadog-agent-cluster-agent-admission-controller, kube_service:datadog-agent-cluster-agent-metrics-api, pod_phase:running, short_image:cluster-agent                                                                      
```

You should also be able to see the metric on the DD UI Metrics Explorer as seen in the figure below:

![image](https://github.com/DataDog/integrations-core/assets/41540817/b3d7a336-4cf0-41ea-9db3-9ada932bef82)


### Review checklist (to be filled by reviewers)
